src/vs/workbench/contrib/chat/browser/actions/chatCodeblockActions.ts

```diff
@@ -188,10 +200,10 @@ abstract class InsertCodeBlockAction extends ChatCodeBlockAction {
 			return;
 		}
 
-		await this.handleTextEditor(accessor, activeEditorControl, context);
+		await this.handleTextEditor(progressService, notificationService, languageFeaturesService, bulkEditService, codeEditorService, chatService, activeEditorControl, context);
 	}
 
-	private async handleNotebookEditor(accessor: ServicesAccessor, notebookEditor: INotebookEditor, context: ICodeBlockActionContext) {
+	private async handleNotebookEditor(languageService: ILanguageService, progressService: IProgressService, notificationService: INotificationService, languageFeaturesService: ILanguageFeaturesService, bulkEditService: IBulkEditService, codeEditorService: ICodeEditorService, chatService: IChatService, notebookEditor: INotebookEditor, context: ICodeBlockActionContext) {
```

Comment:
This is getting kind of crazy, you can use instantiationService.invokeFunction to get a new accessor or create something with injected services.

Reply:
Ah thanks, yeah the async `openEditor` broke all subsequent `accessor.get`s